R/watcher.r

```
@@ -47,11 +47,10 @@ watch <- function(path, callback, pattern = NULL, hash = TRUE) {
 #' @return a digest of the file, or NA if it doesn't exist.
 #' @keywords internal
 safe_digest <- function(path) {
-  if (!file.exists(path)) return(NA_character_)
-  if (is_directory(path)) return(NA_character_)
-  if (!is_readable(path)) return(NA_character_)
-
-  digest::digest(path, file = TRUE)
+  tryCatch(
+    digest::digest(path, file = TRUE),
+    error = function(e) NA_character_
+  )
```

testthat used to use the `tryCatch()` strategy but the problem is that it silently ignored other types of errors apart from file existence.